File: src/main/java/pl/tsimur/jpatest/service/SubscriptionServiceImpl.java
package pl.tsimur.jpatest.service;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;
import pl.tsimur.jpatest.model.Subscription;
import pl.tsimur.jpatest.model.Tariff;
import pl.tsimur.jpatest.repository.SubscriptionRepository;

@Slf4j
@Service
@RequiredArgsConstructor
public class SubscriptionServiceImpl implements SubscriptionService {

    private final SubscriptionRepository subscriptionRepository;

    @Override
    public Subscription find(String name) {
        Subscription subscription =
                subscriptionRepository.findByName(Tariff.valueOf(name));
        log.info("Found subscription with {} name", name);
        return subscription;
    }

    @Override
    public Subscription findTitle(String title) {
        Subscription subscription =
                subscriptionRepository.findByTitle(title);
        log.info("Found subscription with title: \"{}\" ", title);
        return subscription;
    }

    @Override
    public Subscription find(Float price) {
        Subscription subscription =
                subscriptionRepository.findByPrice(price);
        log.info("Found subscription with price {} pln ", price);
        return subscription;
    }

    @Override
    @Transactional
    public void updateTitle(Tariff name, String title) {
        subscriptionRepository.updateByName(name, title);
        log.info("Subscription with name: {} have new title \"{}\"", name.toString(), title);
    }

    @Override
    @Transactional
    public void updatePrice(String title, Float price) {
        subscriptionRepository.updateByTitle(title, price);
        log.info("Subscription with title: {} have new price = {}", title, price);
    }

}

File: src/main/kotlin/pl/timur/jpatest/service/SubscriptionServiceImpl.kt
package pl.timur.jpatest.service

import org.slf4j.LoggerFactory
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional
import pl.timur.jpatest.model.Subscription
import pl.timur.jpatest.model.Tariff
import pl.timur.jpatest.repository.SubscriptionRepository

@Service
class SubscriptionServiceImpl(
    private val subscriptionRepository: SubscriptionRepository
) : SubscriptionService {

    companion object {
        private val log = LoggerFactory.getLogger(this::class.java)
    }

    override fun find(name: String): Subscription? {
        val subscription = subscriptionRepository.findByName(Tariff.valueOf(name))
        log.info("Found subscription with $name name")
        return subscription
    }

    override fun findTitle(title: String): Subscription? {
        val subscription = subscriptionRepository.findByTitle(title)
        log.info("""Found subscription with title: "$title"""")
        return subscription
    }

    override fun find(price: Float): Subscription? {
        val subscription = subscriptionRepository.findByPrice(price)
        log.info("Found subscription with price $price pln ")
        return subscription
    }

    @Transactional
    override fun updateTitle(name: Tariff, title: String) {
        subscriptionRepository.updateByName(name, title)
        log.info("Subscription   with name: ${name.name} have new title; $title")
    }

    @Transactional
    override fun updatePrice(title: String, price: Float) {
        subscriptionRepository.updateByTitle(title, price)
        log.info("Subscription with title: $title have new price = $price")
    }
}